pytorch_lightning/callbacks/quantization.py

```diff
@@ -0,0 +1,201 @@
+    if '.' in attribs:
+        attrib, attribs = attribs.split('.', 1)
+        if hasattr(obj, attrib):
+            return _recursive_hasattr(getattr(obj, attrib), attribs, state)
+        else:
+            return False
+    else:
+        return state & hasattr(obj, attribs)
+
+
+class QuantizationAwareTraining(Callback):
+    """
+    Quantization allows speed-up inference and decrease model size by performing computations and storing tensors
+     at lower bitwidths than floating point precision such as INT8 or FLOAT16.
+    We use native PyTorch API so for more information see
+     `Quantization <https://pytorch.org/docs/stable/quantization.html#quantization-aware-training>_`
+    """
+
+    OBSERVER_TYPES = ('histogram', 'average')
+
+    def __init__(
+        self,
+        qconfig: Union[str, QConfig] = 'fbgemm',
+        observer_type: str = "average",
+        collect_quantization: Optional[Union[int, Callable]] = None,
+        modules_to_fuse: Optional[Sequence] = None,
+        input_compatible: bool = True,
+    ) -> None:
+        """
+        Args:
+            qconfig: define quantization configuration see: `torch.quantization.QConfig
+             <https://pytorch.org/docs/stable/torch.quantization.html?highlight=qconfig#torch.quantization.QConfig>_`
+                or use pre-defined: 'fbgemm' for server inference and 'qnnpack' for mobile inference
+            observer_type: switching between `MovingAverageMinMaxObserver` as "average" (default)
+                and `HistogramObserver` as "histogram" which is more computational costly
+            collect_quantization: define custom count or function when you shall collect quantization statistic
+
+                - with default ``None`` you call the quantization observer for each module forward,
+                    typical use-case can be reflecting user augmentation
+                - custom call count to limit explicit number of calls starting from beginning
+                - custom lambda function with single trainer argument,
+                    see example when you limit call only for last epoch::
+
+                    def custom_trigger_last(trainer):
+                        return trainer.current_epoch == (trainer.max_epochs - 1)
+
+                    QuantizationAwareTraining(lambda_trigger=custom_trigger_last)
+
+            modules_to_fuse: allows you fuse a few layer together as you can see in `diagram
+             <https://pytorch.org/docs/stable/quantization.html#quantization-aware-training>_`
+                to find what layer types can be fue=sed check https://github.com/pytorch/pytorch/pull/43286
+            input_compatible: preserve quent/dequant layers which allows to feat any input as to the original model,
+                but break compatibility to torchscript
+        """
+        if not isinstance(qconfig, (str, QConfig)):
+            raise MisconfigurationException(f"Unsupported qconfig: f{qconfig}.")
+        self._qconfig = qconfig
```

Comment:
check for correct string?
```suggestion
        if isinstance(qconfig, str) and qconfig not in ('fbgemm', 'qnnpack'):
            raise MisconfigurationException(f"Unsupport string for qconfig")
        self._qconfig = qconfig
```

Reply:
not sure if want to limit it here since we just pass the parameter to PT and if P adds an extra argument we would have to hardcode it here...

Reply:
alright, fine by me :]